### ISSUE TITLE
WGAN-GP implementation incompatible with TensorFlow 2.x

### ISSUE THREAD

**Issue by DCtheTall** [davidADSP/GDL_code#67]
FWIW the WGAN-GP implementation in this repo, as the issue title suggests, breaks when you try to use TensorFlow 2.x. When you try to use the current implementation `K.gradients()` returns `None` when computing the gradient penalty loss.

I am not sure if this book is intentionally restricted to TF 1.x, but it might be nice to mention this breakage somewhere?

There's an implementation of a WGAN-GP using Keras and TensorFlow [here](https://github.com/timsainb/tensorflow2-generative-models/blob/master/3.0-WGAN-GP-fashion-mnist.ipynb) (this code has the WGAN-GP extend the `keras.models.Model` class but I don't think that is actually necessary).

**Comment by arisliang** [comment]
Have you tried the `tensorflow_2` branch?
